Parse/PFFile.m

```diff
@@ -73,12 +73,26 @@ + (instancetype)fileWithName:(NSString *)name contentsAtPath:(NSString *)path {
 + (instancetype)fileWithName:(NSString *)name contentsAtPath:(NSString *)path error:(NSError **)error {
     NSFileManager *fileManager = [NSFileManager defaultManager];
     BOOL directory = NO;
-    PFParameterAssert([fileManager fileExistsAtPath:path isDirectory:&directory] && !directory,
-                      @"%@ is not a valid file path for a PFFile.", path);
 
-    NSDictionary *attributess = [fileManager attributesOfItemAtPath:path error:nil];
-    unsigned long long length = [attributess[NSFileSize] unsignedLongValue];
-    PFParameterAssert(length <= PFFileMaxFileSize, @"PFFile cannot be larger than %lli bytes", PFFileMaxFileSize);
+    if (![fileManager fileExistsAtPath:path isDirectory:&directory] || directory) {
+        NSString *message = [NSString stringWithFormat:@"Failed to create PFFile at path '%@': "
+                                                        "file does not exist.", path];
+        *error = [NSError errorWithDomain:PFParseErrorDomain
```

Comment:
2 things here:
- let's use `PFErrorUtilities` to create an error, since it enforces PFParseErrorDomain, as well as logs to console by default.
- We should fine a code that fits this best from existing error codes, or avoid using `PFParseErrorDomain`, since we have a guarantee that we never use anything other than errors defined in PFConstants (`PFErrorCode`) enum.